test/requests/application_gateway/test_check_ag_exists.rb

```diff
@@ -16,16 +16,16 @@ def test_check_ag_exists_success
   end
 
   def test_check_ag_exists_failure
```

Comment:
This test case should check for an exception other than the resource not being found. You can set the value of the code to anything like 'Exception' or 'NetworkException'.
You can create a separate unit test for cases where the resource group or the parent resource does not exist.